app/src/main/java/dev/andrylat/raqimbek/bankingutils/BankingUtilsApp.java

```diff
@@ -1,43 +1,116 @@
 
   public static void main(String[] args) {
     run();
   }
 
   private static void run() {
-    userInteraction.write(false, List.of("Hello. Enter card number for validation:"));
+    var dialogMapOptional = getDialogMap();
 
-    var userInput = userInteraction.read();
-    var cardValidationInfo = cardValidator.checkCardNumber(userInput);
+    if (dialogMapOptional.isPresent()) {
+      var dialogMap = dialogMapOptional.get();
 
-    if (!cardValidationInfo.getIsValid()) {
-      var errors = cardValidationInfo.getErrors();
+      handleSelectedBankingServiceInput(selectBankingService(dialogMap));
+    }
+  }
 
-      userInteraction.write(true, errors);
-    } else {
-      var paymentSystemOptional =
-          paymentSystemDeterminer.determinePaymentSystemByCardNumber(userInput);
-
-      if (paymentSystemOptional.isPresent()) {
-        var paymentSystem = paymentSystemOptional.get();
-        var message =
-            new StringBuilder("Card is valid. Payment System is \"")
-                .append(paymentSystem)
-                .append("\"")
-                .toString();
-
-        userInteraction.write(false, List.of(message));
+  private static void handleSelectedBankingServiceInput(@NonNull Dialog dialog) {
+    dialog.run();
+  }
+
+  @NonNull
+  private static String getBankingServiceSelectionPromptMessage(
+      @NonNull Map<Integer, Dialog> dialogMap) {
+    StringBuilder greetingMessage =
+        new StringBuilder("Hello. Please type the index of the service you need:\n");
+
+    dialogMap.entrySet().stream()
+        .sorted(Map.Entry.comparingByKey())
+        .forEach(
+            e ->
+                greetingMessage
+                    .append("[")
+                    .append(e.getKey())
+                    .append("] - ")
+                    .append(e.getValue())
+                    .append("\n"));
+
+    return greetingMessage.toString();
+  }
+
+  private static Dialog selectBankingService(@NonNull Map<Integer, Dialog> dialogMap) {
+
+    var promptMessage = getBankingServiceSelectionPromptMessage(dialogMap);
+
+    commandLineUserInteraction.write(promptMessage);
+
+    var selectedBankingService = -1;
+
+    do {
+      var input = commandLineUserInteraction.read();
+
+      if (!isValidBankingServiceIndex(input, dialogMap)) {
+        commandLineUserInteraction.write(
+            "Please write only a number representing an index of a service.");
       }
+
+      selectedBankingService = Integer.parseInt(input);
+
+    } while (selectedBankingService < 0);
+
+    return dialogMap.get(selectedBankingService);
+  }
+
+  private static boolean isValidBankingServiceIndex(
+      @NonNull String input, @NonNull Map<Integer, Dialog> dialogMap) {
+    // Regex pattern to match a non-negative integer
+    var regex = "^(0|[1-9]\\d*)$";
+    return input.matches(regex) && dialogMap.containsKey(Integer.parseInt(input));
+  }
+
+  @NonNull
+  private static Optional<Map<Integer, Dialog>> getDialogMap() {
```

Comment:
Why this logic is so complex? 

you could write method with `"return Map.of(1, new XXX(), 2, new YYY());"`